storm/check.py

```diff
@@ -0,0 +1,688 @@
+        intervals = instance.get('intervals', self.init_config.get('intervals', [60]))
+        self.intervals = []
+        if isinstance(intervals, (list, tuple)):
+            for interval in intervals:
+                if isinstance(interval, (list, tuple)):
+                    for i in interval:
+                        self.intervals.append(int(i))
+                else:
+                    self.intervals.append(int(interval))
+        elif isinstance(intervals, (int, long)):
+            self.intervals.append(intervals)
+        else:
+            self.intervals.extend([int(i) for i in intervals.split(',')])
+
+    def check(self, instance):
+        """ Perform the agent check.
+
+        :param instance: Agent instance.
+        :return: None
+        """
+        # Setup
+        self.update_from_config(instance)
+
+        # Cluster Stats
+        cluster_stats = self.get_storm_cluster_summary()
+        for k, v in self.process_cluster_stats(self.environment_name, cluster_stats).items():
+            value = v[0]
+            tags = v[1]
+            self.report_gauge(metric=k, value=value, tags=tags, additional_tags=self.additional_tags)
+
+        # Nimbus Stats
+        nimbus_stats = self.get_storm_nimbus_summary()
+        for ns in self.process_nimbus_stats(self.environment_name, nimbus_stats):
+            for k, v in ns.items():
+                value = v[0]
+                tags = v[1]
+                self.report_gauge(metric=k, value=value, tags=tags, additional_tags=self.additional_tags)
+
+        # Supervisor Stats
+        supervisor_stats = self.get_storm_supervisor_summary()
+        for ss in self.process_supervisor_stats(supervisor_stats):
+            for k, v in ss.items():
+                value = v[0]
+                tags = v[1]
+                self.report_gauge(metric=k, value=value, tags=tags, additional_tags=self.additional_tags)
+
+        # Topology Stats
+        summary = self.get_storm_topology_summary()
+        for topology in summary['topologies']:
+            topology_id = topology.get('id')
+            topology_name = _g(topology, 'unknown', None, 'name')
+            if topology_name not in self.excluded_topologies:
+                stats = self.get_topology_info(topology_id=topology_id)
+                interval_results = []
+                for interval in self.intervals:
+                    results = self.process_topology_stats(topology_stats=stats, interval=interval)
+                    if len(results) > 0:
+                        interval_results.append(results)
+                metric_stats = self.get_topology_metrics(topology_id=topology_id)
+                metric_results = self.process_topology_metrics(topology_name, metric_stats)
+
+                topology_status = _g(stats, 'unknown', None, 'status').upper()
+                if topology_status != 'ACTIVE':
+                    check_status = AgentCheck.CRITICAL
+                    self.service_check(
```

Comment:
The `self.service_check` could be put outside the is/else to avoid duplicating it at line 624 (the `message` could use `topology_status` in every case).